folium/features.py

```diff
@@ -602,25 +603,123 @@ def get_bounds(self):
+    >>> )
+    # Provide fields, with labels off, and sticky True.
+    >>> Tooltip(fields=('CNTY_NM',), labels=False)
     """
-    if isinstance(tooltip, Tooltip):
-        if tooltip.fields:
-            for value in tooltip.fields:
-                assert value in keys, ("The value {} is not available in {}"
-                                       .format(value, keys))
-        return tooltip
-    else:
-        return create_tooltip(tooltip)
+    _template = Template(u"""
+        {% macro script(this, kwargs) %}
+        {{ this._parent.get_name() }}.bindTooltip(
+            function(layer){
+            let fields = {{ this.fields }};
+            {% if this.aliases %}
+            let aliases = {{ this.aliases }};
+            {% endif %}
+            return '<table{% if this.style %} style="{{this.style}}"{% endif%}>' +
+            String(
+                fields.map(
+                columnname=>
+                    `<tr style="text-align: left;">{% if this.labels %}
+                    <th style="padding: 4px; padding-right: 10px;">{% if this.aliases %}
+                        ${aliases[fields.indexOf(columnname)]
+                        {% if this.toLocaleString %}.toLocaleString(){% endif %}}
+                    {% else %}
+                    ${ columnname{% if this.toLocaleString %}.toLocaleString(){% endif %}}
+                    {% endif %}</th>
+                    {% endif %}
+                    <td style="padding: 4px;">${ layer.feature.properties[columnname]
+                    {% if this.toLocaleString %}.toLocaleString(){% endif %}}</td></tr>`
+                ).join(''))
+                +'</table>'
+            }, {{ this.kwargs }});
+        {% endmacro %}
+        """)
+
+    def __init__(self, fields, aliases=None, labels=True,
+                 toLocaleString=False, style=None, sticky=True, **kwargs):
+        super(GeoJsonTooltip, self).__init__(
+            text='', style=style, sticky=sticky, **kwargs
+        )
+        self._name = "Tooltip"
+
+        assert isinstance(fields, (list, tuple)), "Please pass a list or " \
+                                                  "tuple to fields."
+        if aliases is not None:
+            assert isinstance(aliases, (list, tuple))
+            assert len(fields) == len(aliases), "fields and aliases must have" \
+                                                " the same length."
+        assert isinstance(labels, bool), "labels requires a boolean value."
+        assert isinstance(toLocaleString, bool), "toLocaleString must be " \
+                                                 "boolean."
+
+        self.fields = fields
+        self.aliases = aliases
+        self.labels = labels
+        self.toLocaleString = toLocaleString
+        if style:
+            assert isinstance(style, str), \
+                "Pass a valid inline HTML style property string to style."
+            # noqa outside of type checking.
+            self.style = style
+
```

Comment:
Would it be good to have an intermediary step from `**kwargs` to `options` as in `Tooltip`, to do some parameter checks using `parse_kwargs` to ensure valid parameters are being passed? 

For example, `sticky` is being passed `True` right now but has no effect because it is not part of the `kwargs` dict.

On leaflet's Tooltip [options](https://leafletjs.com/reference-1.3.2.html#tooltip), everything should be good to be passed except for `permanent` - this GeoJson tooltip is a different animal because it's a function that evaluates on a mouseover and it's not being directly bound to every feature.

Reply:
Since in `GeoJsonTooltip` the kwargs are passed to regular `Tooltip`, they are validated in there. 

In `Tooltip`, `sticky` is added to kwargs before validating:
`kwargs.update({'sticky': sticky})`

Are there options that work with `Tooltip` but not with `GeoJsonTooltip`? It seems `permanent` indeed does nothing. We can assert that option is not entered in kwargs.

Reply:
@Conengmo `sticky`, or any non-named `kwargs` for that matter, don't seem to be adding to tooltip in `GeoJsonTooltip`: [nbviewer link](http://nbviewer.jupyter.org/gist/jtbaker/6979f6d669a658848bc7eccf44c3af0d)

lines [680-682](https://github.com/Conengmo/folium/blob/tooltip-split/folium/features.py#L680-L682) don't seem to be having an effect. I don't totally understand `super`, but it seems like it's allowing us to escape out to the parent Tooltip class for argument inheritance. 

Don't we still need to pass `self` to the `Tooltip.__init__()` to assign to our GeoJsonConstructor?

Reply:
Figured it out, [line 680-682](https://github.com/Conengmo/folium/blob/tooltip-split/folium/features.py#L680-L682):

`        super(GeoJsonTooltip, self).__init__(
            text='', style=style, sticky=sticky, **kwargs
        )`

needs to switch to:

`        super(GeoJsonTooltip, self).__init__(
            self, style=style, sticky=sticky, **kwargs
        )`

Swapping those out and behavior worked as expected for me. Do you think this will have any ancillary effects?

Reply:
I made a mistake, sorry about that. I forgot to rename the kwargs variable in one place so the options didn't get through. Fixed that now. Now the options in `GeoJsonTooltip` get parsed by its parent.

You're right about what `super` does. You don't have to pass `self`, it does that for you. And in Python 3 you can just call `super().__init__()`, even easier.

Reply:
@Conengmo just discovered that, about `super`, and found the `self.options` issue in the `GeoJsonTooltip` as well. I made a new branch called `tooltip-split` as well, but github doesn't seem to let me PR to your `tooltip-split`.

Incorporated a few new features including a `handleOptions` function to parse non-primitive values in GeoJson properties, and renamed `toLocaleString` to `localize`.

Reply:
I will PR my `tooltip-split` into `patch-2`. It should have your commit history as well.

Reply:
I made a new branch called `tooltip-split` as well, but Github doesn't seem to let me PR to your `tooltip-split`.

Incorporated a few new features including a `handleObject` function to parse non-primitive values in GeoJson properties, and renamed `toLocaleString` to `localize`.

I will PR my tooltip-split into patch-2. It should have your commit history as well.

Reply:
So we fixed the same bug simultaneously! I didn't stand a chance.